src/bos/server/controllers/v2/sessiontemplates.py

```diff
@@ -63,12 +63,50 @@ def _sanitize_xnames(st_json):
     Returns:
       Nothing
     """
-    if 'boot_sets' in st_json:
-        for boot_set in st_json['boot_sets']:
-            if 'node_list' in st_json['boot_sets'][boot_set]:
-                clean_nl = [_canonize_xname(node) for node in
-                            st_json['boot_sets'][boot_set]['node_list']]
-                st_json['boot_sets'][boot_set]['node_list'] = clean_nl
+    # There should always be a boot_sets field -- this function
+    # is only called after the template has been verified
+    for boot_set in st_json['boot_sets'].values():
+        if 'node_list' not in boot_set:
+            continue
+        boot_set['node_list'] = [_canonize_xname(node) for node in boot_set['node_list']]
+
+
+def _validate_sanitize_session_template(session_template_id, template_data):
+    """
+    Used when creating or patching session templates
+    """
+    # The boot_sets field is required.
+    if "boot_sets" not in template_data:
+            raise ParsingException("Missing required 'boot_sets' field")
+
+    """All keys in the boot_sets mapping must match the 'name' fields in the
```

Comment:
Why not use the hash # for comments rather than the multi-line docstring notation? The quotes are legitimate but non-standard. If you are going to use the quotes, why not put a line break after the initial set of quotes?

Reply:
Agreed. I have a follow-up commit to fix this. It is part of my next PR to the feature branch.